orttraining/orttraining/test/training_api/trainer/trainer.cc

```diff
@@ -220,14 +246,15 @@ void RunTraining(const TestRunnerParameters& params) {
       train_step_range.Begin();
 #endif
 
-      std::vector<Ort::Value> fetches;
-      EnforceCheck(module.TrainStep(inputs, fetches), "Failed during module.TrainStep.");
-
+      std::vector<OrtValue*> fetches(train_mode_output_count);
+      ORT_RETURN_ON_ERROR(g_ort_api->TrainStep(session, nullptr,
+                                               inputs.size(), (const OrtValue* const*)inputs.data(),
+                                               train_mode_output_count, fetches.data()));
 #if defined(USE_CUDA) && defined(ENABLE_NVTX_PROFILE)
       train_step_range.End();
 #endif
 
-      float loss = *(fetches[0].GetTensorMutableData<float>());
+      float loss = onnxruntime::training::api::utils::GetValue<float>(*fetches[0]);
```

Comment:
this is the thing I was trying to avoid, because operating on OrtValue will need user code access our non-public APIs, in this case, it is the GetValue.

Reply:
You are right! I will fix this